service/tasks/driver.py

```diff
@@ -1197,13 +1201,35 @@ def check_web_desktop_task(driverCls, provider, identity,
     except AnsibleDeployException as exc:
         check_web_desktop_task.retry(exc=exc)
     except Instance.DoesNotExist:
-        celery_logger.warn("check_web_desktop_task failed: Instance %s no longer exists"
-                           % instance_alias)
+        celery_logger.warn("check_web_desktop_task failed: Instance %s no longer exists" % instance_alias)
     except (BaseException, Exception) as exc:
         celery_logger.exception(exc)
         check_web_desktop_task.retry(exc=exc)
 
 
+@task(name="add_security_group_task",
+      max_retries=5,
+      default_retry_delay=10)
+def add_security_group_task(driverCls, provider, core_identity,
+                            instance_alias, *args, **kwargs):
+    """
+    Assign the security group to the instance using the OpenStack Nova API
+    """
+    from service.instance import _to_user_driver
+    user_driver = _to_user_driver(core_identity)
+    user_nova = user_driver.nova
+    try:
+        server_instance = user_nova.servers.get(instance_alias)
+    except:
+        raise Exception("Cannot find the instance")
+    try:
+        security_group_name = core_identity.provider.get_config("network", "security_group_name", "atmosphere")
+        server_instance.add_security_group(security_group_name)
+    except:
```

Comment:
Does this potentially catch some exception from Nova?  You might want to know which one, or at least only catch the ones you typically expect.  Same for line 1223 above.